pandas/tests/test_reshape.py

```diff
@@ -551,11 +576,23 @@ def test_basic_drop_first_NA(self):
                                       ['b', nan], 1)
         assert_frame_equal(res_na, exp_na)
 
+        res_na = get_dummies(s_NA, dummy_na='keep', sparse=self.sparse,
+                             drop_first=True)
+        exp_na = DataFrame({'b': {0: 0,
```

Comment:
this is awkard constrution, do use a dict-of-dicts, just dict-of-list is much more natural.